### PR TITLE
Add hub stats

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
This PR uses the same metrics from drivers for the hub.

### CODE REVIEW

File: src/stats/actor.rs [hunk 1]
```diff
@@ -47,6 +49,23 @@ impl StatsActor {
             }
         });
 
+        let hub_stats_task = tokio::spawn({
+            let hub = self.hub.clone();
+            let update_period = Arc::clone(&self.update_period);
```
Comment (patrickelectric) [inline src/stats/actor.rs:54]: is there a reason for not calling clone directly ?

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: src/stats/actor.rs (hunk 1 above)
Comment (joaoantoniocardoso) [reply src/stats/actor.rs:54]: there is no difference, that's a more _expressive_ way to saying you are cloning the ARC, not the data, but we can use .clone, no worries!

File: src/stats/driver.rs [hunk 2]
```diff
@@ -54,8 +54,10 @@ impl Default for AccumulatedStatsInner {
...
     pub async fn update(&mut self, message: Arc<Protocol>) {
         self.last_update = chrono::Utc::now().timestamp_micros() as u64;
-        self.bytes += message.raw_bytes().len();
-        self.messages += 1;
-        self.delay += self.last_update - message.timestamp;
+        let _ = self.bytes.wrapping_add(message.raw_bytes().len() as u64);
+        let _ = self.messages.wrapping_add(1);
+        let _ = self
+            .delay
+            .wrapping_add(self.last_update - message.timestamp);
```
Comment (joaoantoniocardoso) [inline src/stats/driver.rs:61]: lol, I got confused with the function notation

```suggestion
        self.bytes = self.bytes.wrapping_add(message.raw_bytes().len() as u64);
        self.messages = self.messages.wrapping_add(1);
        self.delay = self
            .delay
            .wrapping_add(self.last_update - message.timestamp);
```

File: src/stats/actor.rs (hunk 1 above)
Comment (joaoantoniocardoso) [reply src/stats/actor.rs:54]: ... or just let's use it :laughing: 

image

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**